rs-backend/src/git.rs

```diff
@@ -0,0 +1,59 @@
+use std::env;
+use std::path::Path;
+use std::process::Command;
+
+const CHECKOUT_OPTION: &'static str = "master";
+const GIT: &'static str = "git";
+const GIT_CHECKOUT: &'static str = "checkout";
+const GIT_PULL: &'static str = "pull";
+
+
+pub fn get_repo_path() -> Result<String, &'static str> {
+    if env::args().count() < 2 {
+        return Err("No argument supplied, needs location of data repo.");
+    }
+    Ok(env::args().nth(1).unwrap())
+}
+
+pub fn update_repo(repo_path: String) {
+    let working_dir = Path::new(&repo_path);
+
+    if checkout_master(working_dir).is_ok() {
+        if pull_updates(working_dir).is_ok() {
+            println!("Success.");
+        } else {
+            println!("Pull failed.");
+            return;
+        }
+    } else {
+        println!("Checkout failed. Invalid repo path specified, maybe?");
+        return;
+    }
+
+}
+
+fn checkout_master(working_dir: &Path) -> Result<(), &'static str> {
+    if let Ok(checkout) = Command::new(GIT)
+        .current_dir(working_dir)
+        .arg(GIT_CHECKOUT)
+        .arg(CHECKOUT_OPTION)
+        .status() {
+        if checkout.success() {
+            return Ok(());
+        }
+    }
+    Err("Checkout failed.")
```

Comment:
We use error-chain already, so we can just `use errors::*` at the top of this module and return the actual error here. I think you'll need to add `mod git` to the `main.rs` file to import errors. Once you've imported the errors module the return type should be `Result<()>`.